Demo/Demo/Networking/DemoMerchantAPI.swift

```diff
@@ -67,4 +70,36 @@ final class DemoMerchantAPI {
     private func buildBaseURL(with endpoint: String) -> URL? {
         URL(string: DemoSettings.environment.baseURL + endpoint)
     }
+    
+    public func getAccessToken(clientId: String, environment: PaymentsCore.Environment) async -> String?{
+        guard let token = self.accessToken else{
+            self.accessToken = await fetchAccessToken(clientId: clientId, environment: environment)
+            return self.accessToken
+        }
+        return token
+    }
+    
+    private func fetchAccessToken(clientId: String, environment: PaymentsCore.Environment) async -> String? {
+        do {
+            let accessTokenRequest = AccessTokenRequest(clientID: clientId)
+            guard let request = accessTokenRequest.toURLRequest(environment: environment) else {
+                throw URLResponseError.dataParsingError
+            }
+            let (data, response) = try await URLSession.shared.performRequest(with: request)
+            guard let response = response as? HTTPURLResponse else {
+                throw URLResponseError.networkConnectionError
+            }
+            switch response.statusCode {
+                case 200..<300:
+                    let accessTokenResponse: AccessTokenResponse = try parse(from: data)
+                    return accessTokenResponse.accessToken
+            default: throw URLResponseError.serverError
+            }
+        }
+        catch{
+            print("Error in fetching token")
+            return nil
```

Comment:
I think we should throw an error here vs printing a message and returning `nil`

Reply:
Eh, thinking about it now this is the demo app so maybe not as important? Up to you. 😄